src/ui/handler/scroll_zoom.js

```diff
@@ -154,6 +168,10 @@ class ScrollZoomHandler {
 
     wheel(e: WheelEvent) {
         if (!this.isEnabled()) return;
+        if (this.isControlled() && !e.ctrlKey && !e.metaKey) {
+            const scrollZoomControl = new mapboxgl.ScrollZoomControl({closeButton: false}).setHTML();
```

Comment:
A [quick search for addControl](https://github.com/mapbox/mapbox-gl-js/search?q=addControl+path%3Asrc%2F) suggests that `AttributionControl` is the only control automatically added to the map. Do you think this is best as an opt-in control the developer must add or as an opt-out control you'd have to remove?

Reply:
+1 to this, I think its better to have an opt-in Control for this and not clutter the Map constructor params more. So it might be nice to add this in like this
```
const map = new mapboxgl.Map({..});

map.addControl(new mapboxgl.ZoomBlockerControl({}));
```

The reason I think this would be more ergonomic is because it would be easier for users to customize it. With something like this.

```
const map = new mapboxgl.Map({..});
const zoomBlocker = new mapboxgl.ScrollBlockerControl({..})
           .setHTML(`<b>whatever content the user wants</b>`);
map.addControl(zoomBlocker);
```


Then instead of having this logic here the blocking of the zoom event can be done in the `onAdd` method of the control, this way the rest of the codebase is unaware of the blocker control running.